<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="planar_2dof">

  <xacro:arg name="L1" default="0.25" />
  <xacro:arg name="L2" default="0.20" />

  <xacro:property name="L1" value="$(arg L1)" />
  <xacro:property name="L2" value="$(arg L2)" />

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <origin xyz="0 0 0.025" rpy="0 0 0"/>
      <material name="grey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>

  <link name="link1">
    <visual>
      <geometry>
        <box size="${L1} 0.04 0.04"/>
      </geometry>
      <origin xyz="${L1/2} 0 0" rpy="0 0 0"/>
      <material name="orange">
        <color rgba="1.0 0.5 0.0 1.0"/>
      </material>
    </visual>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.05" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-3.14159" upper="3.14159" effort="10.0" velocity="1.0"/>
  </joint>

  <link name="link2">
    <visual>
      <geometry>
        <box size="${L2} 0.04 0.04"/>
      </geometry>
      <origin xyz="${L2/2} 0 0" rpy="0 0 0"/>
      <material name="blue">
        <color rgba="0.1 0.1 1.0 1.0"/>
      </material>
    </visual>
  </link>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="${L1} 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit lower="-3.14159" upper="3.14159" effort="10.0" velocity="1.0"/>
  </joint>

  <link name="ee_link"/>

  <joint name="joint2_to_ee" type="fixed">
    <parent link="link2"/>
    <child link="ee_link"/>
    <origin xyz="${L2} 0 0" rpy="0 0 0"/>
  </joint>

</robot>